sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructField.scala

```diff
@@ -24,7 +24,9 @@ import org.json4s.JsonDSL._
  * A field inside a StructType.
  * @param name The name of this field.
  * @param dataType The data type of this field.
- * @param nullable Indicates if values of this field can be `null` values.
+ * @param nullable Indicates if values of this field can be `null` values. Note that this is ONLY A HINT to the 
```

Comment:
According to @marmbrus's [comment here](https://issues.apache.org/jira/browse/SPARK-11319?focusedCommentId=15014678&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-15014678) this is actually much stronger than a hint:

> Fair, this is **a contract with the optimizer that you will not produce null data**. If you are not certain you will not produce null values then set this to true.